app/store/fetching_thread_store.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {BehaviorSubject} from 'rxjs';
+
+type State = {[id: string]: boolean};
+
+const defaultState: State = {};
+
+export const subject: BehaviorSubject<State> = new BehaviorSubject(defaultState);
+
+export const setFetchingThreadState = (rootId: string, isFetching: boolean) => {
+    const prevState = subject.value;
+    subject.next({
+        ...prevState,
+        [rootId]: isFetching,
```

Comment:
Probably not a big deal, but there is a remote possibility that the same rootId happens in different servers. 0/5 on whether it makes sense to divide this by serverUrl.

Reply:
I don't think that is needed, unless until now, there is no way to see a thread from a server different than the current server